File: android/src/main/kotlin/com/sinon/moja_app_auth/MojaAppAuthPlugin.kt
package com.yourcompany.moja_app_auth

import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.net.Uri
import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class MojaAppAuthPlugin: FlutterPlugin, MethodChannel.MethodCallHandler {
    private lateinit var channel : MethodChannel
    private lateinit var context : Context

    override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(binding.binaryMessenger, "moja_app_auth")
        channel.setMethodCallHandler(this)
        context = binding.applicationContext
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "isMojaAppInstalled" -> {
                val pm: PackageManager = context.packageManager
                try {
                    pm.getPackageInfo("com.sinon.app.moja_app", 0) // ← MojaApp package name
                    result.success(true)
                } catch (e: Exception) {
                    result.success(false)
                }
            }
            "loginWithMojaApp" -> {
                val intent = Intent(Intent.ACTION_VIEW, Uri.parse("moja://login"))
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
                context.startActivity(intent)
                result.success("OTP/TOKEN placeholder from Android")
            }
            "redirectToStore" -> {
                val intent = Intent(Intent.ACTION_VIEW,
                    Uri.parse("market://details?id=com.sinon.app.moja_app"))
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
                context.startActivity(intent)
                result.success(null)
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: ios/Classes/MojaAppAuthPlugin.swift
import Flutter
import UIKit

public class MojaAppAuthPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "moja_app_auth", binaryMessenger: registrar.messenger())
    let instance = MojaAppAuthPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "isMojaAppInstalled":
      let url = URL(string: "moja://login")!
      result(UIApplication.shared.canOpenURL(url))
    case "loginWithMojaApp":
      if let url = URL(string: "moja://login") {
        UIApplication.shared.open(url)
        result("OTP/TOKEN placeholder from iOS")
      } else {
        result(nil)
      }
    case "redirectToStore":
      if let url = URL(string: "itms-apps://itunes.apple.com/app/com.sinon.app.moja_app") {
        UIApplication.shared.open(url)
      }
//       if let url = URL(string: "itms-apps://itunes.apple.com/app/idYOUR_APP_ID") {
//               UIApplication.shared.open(url)
//             }

      result(nil)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
